From b5cae64cd6a3b6ef641a76639f5241e1dff2f4a8 Mon Sep 17 00:00:00 2001
From: Igor Opaniuk <igor.opaniuk@foundries.io>
Date: Tue, 20 Jul 2021 17:55:02 +0300
Subject: [PATCH 094/233] [FIO internal] imx8qm_mek: adjust loadaddr and
 scriptaddr

Adjust scriptaddr for all IMX8QM_MEK parts to 0x83100000.

Set loadaddr for TARGET_IMX8QM_MEK_A72_ONLY to custom 0xC7000000.

Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>
Co-developed-by: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>
Signed-off-by: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>
---
 include/configs/imx8qm_mek.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/configs/imx8qm_mek.h b/include/configs/imx8qm_mek.h
index ed5c179fc75..68005cb24a1 100644
--- a/include/configs/imx8qm_mek.h
+++ b/include/configs/imx8qm_mek.h
@@ -170,6 +170,7 @@
 	XEN_BOOT_ENV \
 	JAILHOUSE_ENV\
 	AHAB_ENV \
+	"scriptaddr=0x83100000\0" \
 	"script=boot.scr\0" \
 	"image=Image\0" \
 	SPLASH_IMAGE_ADDR \
@@ -256,6 +257,8 @@
 
 /* Link Definitions */
 #ifdef CONFIG_TARGET_IMX8QM_MEK_A72_ONLY
+	#undef CONFIG_SYS_LOAD_ADDR
+	#define CONFIG_SYS_LOAD_ADDR		0xC7000000
 	#define CONFIG_SYS_INIT_SP_ADDR		0xC0200000
 #else
 	#define CONFIG_SYS_INIT_SP_ADDR		0x80200000
-- 
2.25.1

